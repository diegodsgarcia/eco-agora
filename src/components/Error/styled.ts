import styled from 'styled-components'

export const Wrapper = styled.div`
  text-align: center;
`

export const Title = styled.h2`
  font-size: 2rem;
`

export const Description = styled.p`
  font-size: 1.8rem;
  margin: 2rem 0;
`
