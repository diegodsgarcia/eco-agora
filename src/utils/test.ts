import { LocaleAPI } from 'types/Locale'
import { buildLocale, convertDate } from 'utils'

test('Should build Locale', () => {
  const mock: LocaleAPI = {
    Cameras: [
      {
        DisplayName: 'Anchieta',
        ExternalCode: 'km 102',
        Cameras: [
          {
            CameraID: 1,
            KM: 12,
            Description: 'km 102',
            LastUpdate: 'Date(1212334324)',
            Url: ''
          }
        ]
      }
    ]
  }

  expect(buildLocale(mock)).toStrictEqual([
    {
      name: mock.Cameras[0].DisplayName,
      externalCode: mock.Cameras[0].ExternalCode,
      cameras: [
        {
          id: mock.Cameras[0].Cameras[0].CameraID,
          km: mock.Cameras[0].Cameras[0].KM,
          description: mock.Cameras[0].Cameras[0].Description,
          lastUpdate: mock.Cameras[0].Cameras[0].LastUpdate,
          url: mock.Cameras[0].Cameras[0].Url
        }
      ]
    }
  ])
})

test('Should build void Locale', () => {
  const mock: LocaleAPI = {
    Cameras: []
  }

  expect(buildLocale(mock)).toStrictEqual([])
})

test('Should be convert to Date', () => {
  const mock = 'Date(1609259906000)'

  expect(convertDate(mock)).toStrictEqual('29/12/2020 13:38:26')
})
