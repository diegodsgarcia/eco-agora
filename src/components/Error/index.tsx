import * as S from './styled'

export type ErrorProps = {
  message: string
}

const Error = ({ message }: ErrorProps) => (
  <S.Wrapper>
    <S.Title>Ops! Ocorreu algum :(</S.Title>
    <S.Description>{message}</S.Description>
  </S.Wrapper>
)

export default Error
