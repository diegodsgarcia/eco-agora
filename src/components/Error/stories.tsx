import { Story, Meta } from '@storybook/react/types-6-0'

import Error, { ErrorProps } from '.'

export default {
  title: 'Error',
  component: Error,
  args: {
    message: 'Ops! Parece que ocorreu algum problema'
  }
} as Meta

export const Default: Story<ErrorProps> = (args) => <Error {...args} />
