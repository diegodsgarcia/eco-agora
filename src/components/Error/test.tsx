import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helper'

import Error from '.'

test('Should render the component', () => {
  const message = 'Ops! Ocorreu algum problema'
  const { container } = renderWithTheme(<Error message={message} />)

  expect(screen.getByText(/ops! ocorreu algum :\(/i)).toBeInTheDocument()
  expect(screen.getByText(message)).toBeInTheDocument()

  expect(container.firstChild).toMatchSnapshot()
})
