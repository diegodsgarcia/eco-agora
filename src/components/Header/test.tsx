import { fireEvent, screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helper'

import Header, { HeaderProps } from '.'
import theme from 'styles/theme'

const props: HeaderProps = {
  menuItems: [
    {
      name: 'Endereço 1',
      externalCode: 'end-1',
      cameras: []
    },
    {
      name: 'Endereço 2',
      externalCode: 'end-2',
      cameras: []
    },
    {
      name: 'Endereço 3',
      externalCode: 'end-3',
      cameras: []
    }
  ],
  onMenuClick: jest.fn()
}

describe('<Header />', () => {
  it('should render header', () => {
    const { container } = renderWithTheme(<Header {...props} />)

    expect(
      screen.getByRole('heading', { name: /eco agora/i })
    ).toBeInTheDocument()

    expect(screen.getByLabelText(/menu button/i)).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should toggle navigation when click in menu button', () => {
    renderWithTheme(<Header {...props} />)

    const nav = screen.getByRole('navigation', { hidden: true })
    const menu = screen.getByLabelText(/menu button/i)

    expect(nav.getAttribute('aria-hidden')).toBe('true')
    expect(nav).toHaveStyle({ opacity: 0 })

    fireEvent.click(menu)

    expect(nav.getAttribute('aria-hidden')).toBe('false')
    expect(nav).toHaveStyle({ opacity: 1 })

    fireEvent.click(screen.getByText(/endereço 1/i))

    expect(props.onMenuClick).toHaveBeenCalledTimes(1)
  })

  it('should render a selected menu item', () => {
    renderWithTheme(<Header {...props} itemSelected={props.menuItems[0]} />)

    expect(screen.getByText(/endereço 1/i)).toHaveStyle({
      color: theme.colors.primary
    })
  })
})
