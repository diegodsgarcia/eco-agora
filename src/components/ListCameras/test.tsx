import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/test/helper'
import { css } from 'styled-components'

import ListCameras from '.'
import * as S from './styled'
import camerasMock from './mock'

describe('<ListCameras />', () => {
  it('should render list of cameras', async () => {
    renderWithTheme(<ListCameras cameras={camerasMock} />)

    expect(screen.getAllByRole('img')).toHaveLength(6)

    expect(screen.queryAllByRole('figure')[0]).toHaveStyleRule('opacity', '1', {
      modifier: `:hover ${css`
        ${S.Description}
      `}`
    })
  })
})
