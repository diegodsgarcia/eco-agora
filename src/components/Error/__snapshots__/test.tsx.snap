// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render the component 1`] = `
.c0 {
  text-align: center;
}

.c1 {
  font-size: 2rem;
}

.c2 {
  font-size: 1.8rem;
  margin: 2rem 0;
}

<div
  class="c0"
>
  <h2
    class="c1"
  >
    Ops! Ocorreu algum :(
  </h2>
  <p
    class="c2"
  >
    Ops! Ocorreu algum problema
  </p>
</div>
`
